### ISSUE TITLE
Resolver that runs an external command or script and uses its stdout as the value

### ISSUE THREAD

**Issue by sammck** [rioj7/command-variable#45]
Would it violate some core principle to add a resolver that shells out to a command and uses the output of the command as the variable value? This would allow use of a keychain, database query, git configuration, AWS configuration, etc. to provide the value.

Would a pull request that adds this resolver be useful?

**Comment by rioj7** [comment]
I think they are working on a way to let an extension read the terminal content or somehow hook into the stdin/stdout. I don't know the status of this feature.

You suggest that somehow the `${input:name}` variable contains a shell command and that the stdout of the command is the result of the `${input:name}` variable?

In issue #31 is the request to use the result of a task, also a shell command, in a pickString command.

If you can make this a pull request it would be nice.

**Comment by sammck** [comment]
Great, I'll look into it. 

What I'm thinking of is the output of a non interactive script or command that does not need a terminal, and which can safely run every time the variable is expanded. Though I can see the use for an interactive script that could prompt the user, it's not what I need personally. 

Perhaps the output of a task would be the cleanest way to achieve it.

**Comment by sammck** [comment]
Yes, in combination with useFirstResult, this seems to address my needs, though a bit more verbosely. Thank you...

**Comment by rioj7** [comment]
@tcm0116 The fact that I do not support the `${input}` variable is that I don't know which `.json` is the source of the command (task, launch, keybindings).

I'll look into the Task Shell Input extension how they do it and add it to this extension.

**Comment by rioj7** [comment]
@tcm0116 You can use the `${command}` variable. And use `rememberTransformed` to store/remember the result of the command.

```json
    "inputs": [
        {
            "id": "pickBazelTarget",
            "type": "command",
            "command": "extension.commandvariable.pickStringRemember",
            "args": {
                "description": "Choose a target",
                "key": "selectedBazelTarget",
                "rememberTransformed": true,
                "options": [
                    [ "Previous Target", "${remember:selectedBazelTarget}" ],
                    [ "Targets", "${command:bazelTargets}" ],
                ],
               "command": {
                 "bazelTargets": {
                   "command": "shellCommand.execute",
                   "args": {
                     "command": "echo '//...' && bazel query 'kind(cc_*, //...)'",
                     "cwd": "${workspaceFolder}"
                   }
                 }
               }
            }
        }
    ]
```

**Comment by tcm0116** [comment]
@rioj7 - that did the trick! Is there a way to resolve the "${remember:selectedBazelTarget}" value when it is displayed in the pciker? Currently, the picker shows "Previous Target ${remember:selectedBazelTarget}", but it would be much more helpful to the user if it could show the value stored in "${remember:selectedBazelTarget}" instead of just "${remember:selectedBazelTarget}".

**Comment by rioj7** [comment]
@tcm0116 Created a separate issue

**Comment by rioj7** [comment]
@sammck  Does the example in this [comment](https://github.com/rioj7/command-variable/issues/45#issuecomment-1416781180) solve your issue?
